app/authentication.js

```diff
@@ -1,5 +1,30 @@
 import User from './models/user';
 import OAuth2Strategy from 'passport-google-oauth2';
+import FacebookStrategy from 'passport-facebook';
+
+var authCallback = function(accessToken, refreshToken, profile, done) {
+  process.nextTick(function() {
+    User.findOne({"$or": [
```

Comment:
This line means find a user whose facebookId matches profile.id OR whose googleId matches profile.id OR whose email is in profile.emails